### ISSUE TITLE
Detect observable issues

### ISSUE THREAD

**Issue by SamVerschueren** [SamVerschueren/listr#31]
Due to an issue in [is-observable](https://github.com/sindresorhus/is-observable/issues/1), we have to fall back to duck typing as `isObservable()` always returns `false`.

The current [implementation](https://github.com/SamVerschueren/listr/blob/master/lib/utils.js#L5).

```js
const isObservable = obj => Boolean(obj && typeof obj.subscribe === 'function' && obj.constructor.name === 'Observable');
```

Apparently it's not entirely correct

From @TylorS 

> This actually breaks the stuff that I'm working on because of the changes to isObservable. The stream library I'm using is [most.js](https://github.com/cujojs/most) and it's constructor.name is Stream and not Observable. However it does implement Observable interfaces.

From @andywer 

> So how about that: We check with the current custom logic and fallback to is-observable or vice versa. If one of both is true we can be almost certain it is an Observable.
> 
> PS: I can also open an issue on zen-observable to use the symbol-observable so is-observable will work with the zen-observable as well.

// @jamestalmage @sindresorhus @jfmengels

**Comment by SamVerschueren** [comment]
@andywer The problem with `is-observable` is not that it doesn't work with `zen-observable`, it just gives us [false-negatives](https://github.com/sindresorhus/is-observable/issues/1#issuecomment-214247101) and this has something to do with the [import order](https://github.com/sindresorhus/is-observable/issues/1#issuecomment-214479549). We can't change the import order because the Observable is provided by the dev that uses `listr`.

**Comment by andywer** [comment]
@SamVerschueren Right, I forgot about the import order issue (https://github.com/sindresorhus/is-observable/issues/1#issuecomment-214479549)

Drats... 🙈
